### PR TITLE
Make MongoDB extension default artifact a shaded one [HZ-3660, HZ-3635]

### LINKED ISSUE #25895: com.hazelcast.mapstore.* 'NoClassDefFoundError' [HZ-3660]
_master_ (commit 0d36189619cd30230e4a68818658c2183c427337)
Failed on oracle-11: https://jenkins.hazelcast.com/job/Hazelcast-master-postgres-16/2/testReport/junit/com.hazelcast.mapstore.postgres/

<details><summary>Example Stacktrace:</summary>

```
java.lang.NoClassDefFoundError: com/hazelcast/shaded/org/apache/calcite/rex/RexVisitor
	at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3137)
	at java.base/java.lang.Class.getConstructor0(Class.java:3342)
	at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2553)
	at com.hazelcast.internal.util.ServiceLoader$NewInstanceIterator.next(ServiceLoader.java:293)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at com.hazelcast.jet.sql.impl.connector.SqlConnectorCache.<init>(SqlConnectorCache.java:38)
	at com.hazelcast.jet.sql.impl.CalciteSqlOptimizer.<init>(CalciteSqlOptimizer.java:248)
	at com.hazelcast.sql.impl.SqlServiceImpl.start(SqlServiceImpl.java:95)
	at com.hazelcast.spi.impl.NodeEngineImpl.start(NodeEngineImpl.java:292)
	at com.hazelcast.instance.impl.Node.start(Node.java:483)
	at com.hazelcast.instance.impl.HazelcastInstanceImpl.<init>(HazelcastInstanceImpl.java:125)
	at com.hazelcast.instance.impl.HazelcastInstanceFactory.constructHazelcastInstance(HazelcastInstanceFactory.java:218)
	at com.hazelcast.instance.impl.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:197)
	at com.hazelcast.test.TestHazelcastInstanceFactory.newHazelcastInstance(TestHazelcastInstanceFactory.java:236)
	at com.hazelcast.jet.SimpleTestInClusterSupport.initializeWitSupplier(SimpleTestInClusterSupport.java:80)
	at com.hazelcast.jet.SimpleTestInClusterSupport.initialize(SimpleTestInClusterSupport.java:68)
	at com.hazelcast.jet.sql.impl.connector.jdbc.JdbcSqlTestSupport.initialize(JdbcSqlTestSupport.java:78)
	at com.hazelcast.jet.sql.impl.connector.jdbc.JdbcSqlTestSupport.initialize(JdbcSqlTestSupport.java:65)
	at com.hazelcast.mapstore.postgres.PostgresGenericMapStoreTest.beforeClass(PostgresGenericMapStoreTest.java:30)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:41)
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.ClassNotFoundException: com.hazelcast.shaded.org.apache.calcite.rex.RexVisitor
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 25 more

```

</details>

<details><summary>Standard output:</summary>

```
11:03:58,970  INFO || - [MetricsConfigHelper] Time-limited test - [LOCAL] [dev] [5.4.0-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
11:03:58,972  INFO || - [logo] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
11:03:58,972  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
11:03:58,972  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Hazelcast Platform 5.4.0-SNAPSHOT (20231102 - 0d36189) starting at [127.0.0.1]:5701
11:03:58,972  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Cluster name: dev
11:03:58,972  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
11:03:58,973  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Jet is enabled
11:03:58,983  INFO || - [MetricsConfigHelper] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
11:03:58,984  INFO || - [TpcServerBootstrap] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] TPC: disabled
11:03:58,998  WARN || - [CPSubsystem] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
11:03:59,019  INFO || - [JetServiceBackend] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
11:03:59,055  WARN || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Terminating forcefully...
11:03:59,055  INFO || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Shutting down connection manager...
11:03:59,055  INFO || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Shutting down node engine...
11:03:59,061  INFO || - [NodeExtension] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Destroying node NodeExtension.
11:03:59,061  INFO || - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Hazelcast Shutdown is completed in 6 ms.
11:03:59,577  INFO || - [SimpleTestInClusterSupport] Time-limited test - Terminating instance factory in SimpleTestInClusterSupport.@AfterClass

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-postgres-16/2/

[com.hazelcast.mapstore.postgres.PostgreGenericMapStoreIntegrationTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-postgres-16/lastSuccessfulBuild/testReport/junit/com.hazelcast.mapstore.postgres/PostgreGenericMapStoreIntegrationTest/___/)
[com.hazelcast.mapstore.postgres.PostgresGenericMapStoreTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-postgres-16/lastSuccessfulBuild/testReport/junit/com.hazelcast.mapstore.postgres/PostgresGenericMapStoreTest/___/)
[com.hazelcast.mapstore.mysql.MySQLGenericMapStoreIntegrationTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-mysql-8.2/lastSuccessfulBuild/testReport/junit/com.hazelcast.mapstore.mysql/MySQLGenericMapStoreIntegrationTest/___/)
[com.hazelcast.mapstore.mysql.MySQLGenericMapStoreTest.(?)](https://jenkins.hazelcast.com/job/Hazelcast-master-mysql-8.2/lastSuccessfulBuild/testReport/junit/com.hazelcast.mapstore.mysql/MySQLGenericMapStoreTest/___/)

### PR DESCRIPTION
**Pull request by TomaszGaweda**
Fixes #25895

EE PR: https://github.com/hazelcast/hazelcast-enterprise/pull/6766

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Add `Add to Release Notes` label if changes should be mentioned in release notes or `Not Release Notes content` if changes are not relevant for release notes
- [x] Request reviewers if possible
- [ ] New public APIs have `@Nonnull/@Nullable` annotations
- [ ] New public APIs have `@since` tags in Javadoc
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases

### CODE REVIEW

**Comment by TomaszGaweda** [comment]
run-lab-run

**Comment by TomaszGaweda** [comment]
run-lab-run

**Comment by TomaszGaweda** [comment]
run-ee-tests

**Comment by TomaszGaweda** [comment]
run-sonar

File: extensions/mongodb/pom.xml [hunk 1]
```diff
@@ -124,6 +123,12 @@
             <artifactId>calcite-core</artifactId>
             <version>${calcite.version}</version>
             <scope>provided</scope>
+            <exclusions>
```
Comment (ldziedziul) [inline extensions/mongodb/pom.xml:126]: Why this exclusion?

File: extensions/mongodb/pom.xml [hunk 2]
```diff
@@ -48,40 +48,39 @@
...
-                                <relocation>
-                                    <pattern>org.apache.calcite</pattern>
-                                    <shadedPattern>${relocation.root}.org.apache.calcite</shadedPattern>
-                                </relocation>
-                            </relocations>
-                        </configuration>
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
```
Comment (ldziedziul) [inline extensions/mongodb/pom.xml:79]: Do we need this entry, what's wrong with the config from the root pom.xml?

File: extensions/mongodb/pom.xml (hunk 1 above)
Comment (TomaszGaweda) [reply extensions/mongodb/pom.xml]: ups, leftover. 0cbc67e79fcd57a52c3b880b8c2c0513eb75dc42

File: extensions/mongodb/pom.xml (hunk 2 above)
Comment (TomaszGaweda) [reply extensions/mongodb/pom.xml]: TBH I don't exactly know why I added it, I tried many different options - I removed it as indeed it's not needed in the final version

**Comment by TomaszGaweda** [comment]
run-ee-tests

**Comment by TomaszGaweda** [comment]
run-nightly-tests

**Review by ldziedziul — APPROVED** [review]

**Comment by TomaszGaweda** [comment]
Nightly test failures are not connected:
```
[com.hazelcast.internal.networking.nio.iobalancer.IOBalancerMemoryLeakTest.testMemoryLeak_with_RestConnections](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.internal.networking.nio.iobalancer/IOBalancerMemoryLeakTest/testMemoryLeak_with_RestConnections/)
[com.hazelcast.client.tpc.ClientAsyncSocketReaderTest.testValidProtocolBytes_whenSentPartByPart](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.client.tpc/ClientAsyncSocketReaderTest/testValidProtocolBytes_whenSentPartByPart/)
[com.hazelcast.client.tpc.ClientAsyncSocketReaderTest.testAuth_withNotFoundUuid](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.client.tpc/ClientAsyncSocketReaderTest/testAuth_withNotFoundUuid/)
[com.hazelcast.client.tpc.ClientAsyncSocketReaderTest.testMessageSizeLimits_afterAuth](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.client.tpc/ClientAsyncSocketReaderTest/testMessageSizeLimits_afterAuth/)
[com.hazelcast.client.tpc.ClientAsyncSocketReaderTest.testProtocolBytesAndAuth](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.client.tpc/ClientAsyncSocketReaderTest/testProtocolBytesAndAuth/)
[com.hazelcast.client.tpc.ClientTpcTest.testTPCChannelTargetedPendingInvocations_whenConnectionCloses](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.client.tpc/ClientTpcTest/testTPCChannelTargetedPendingInvocations_whenConnectionCloses/)
[com.hazelcast.client.tpc.ClientTpcTest.testTpcClient_heartbeatsToIdleTpcChannels](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.client.tpc/ClientTpcTest/testTpcClient_heartbeatsToIdleTpcChannels/)
[com.hazelcast.client.tpc.ClientTpcTest.testConnectionCloses_whenTpcChannelsClose](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.client.tpc/ClientTpcTest/testConnectionCloses_whenTpcChannelsClose/)
[com.hazelcast.client.tpc.ClientTpcTest.testTPCClient_heartbeatsToNotRespondingTPCChannelsTimeouts](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.client.tpc/ClientTpcTest/testTPCClient_heartbeatsToNotRespondingTPCChannelsTimeouts/)
[com.hazelcast.client.tpc.ClientTpcTest.testClientConnectsAllTpcPorts](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.client.tpc/ClientTpcTest/testClientConnectsAllTpcPorts/)
[com.hazelcast.client.tpc.ClientTpcTest.testClientConnectsAllTpcPorts_whenNewMemberJoins](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.client.tpc/ClientTpcTest/testClientConnectsAllTpcPorts_whenNewMemberJoins/)
[com.hazelcast.client.tpc.ClientTpcTest.testClientRoutesNonPartitionBoundRequestsToClassicConnections](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.client.tpc/ClientTpcTest/testClientRoutesNonPartitionBoundRequestsToClassicConnections/)
[com.hazelcast.client.tpc.ClientTpcTest.testClientRoutesPartitionBoundRequestsToTpcConnections](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.client.tpc/ClientTpcTest/testClientRoutesPartitionBoundRequestsToTpcConnections/)
[com.hazelcast.client.tpc.ClientTpcTest.testClientConnectsAllTpcPorts_afterRestart](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.client.tpc/ClientTpcTest/testClientConnectsAllTpcPorts_afterRestart/)
[com.hazelcast.client.tpc.ClientTpcTest.testPartitionBoundPendingInvocations_whenConnectionCloses](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.client.tpc/ClientTpcTest/testPartitionBoundPendingInvocations_whenConnectionCloses/)
[com.hazelcast.client.tpc.ClientTpcTest.testTpcChannelsClose_whenConnectionCloses](https://jenkins.hazelcast.com/job/Hazelcast-pr-nightly-tests/123/testReport/junit/com.hazelcast.client.tpc/ClientTpcTest/testTpcChannelsClose_whenConnectionCloses/)
```

**Comment by TomaszGaweda** [comment]
run-ee-tests

**Comment by TomaszGaweda** [comment]
run-lab-run

**Comment by TomaszGaweda** [comment]
run-ee-tests

**Comment by TomaszGaweda** [comment]
run-ee-tests

**Comment by TomaszGaweda** [comment]
run-ee-tests

**Review by frant-hartm — APPROVED** [review]

**Pull request merged**